packages/connect/src/protocol/invoke-implementation.spec.ts

```diff
@@ -0,0 +1,279 @@
+          res.header.set("Key", "baz");
+          return {
+            ...res,
+            message: createAsyncIterable(responses),
+          } as typeof res;
+        },
+        (next) => async (req) => {
+          expect(req.header.get("Key")).toEqual("bar");
+          return next(req);
+        },
+      ],
+    )(createAsyncIterable([new Int32Value({ value: 1 })]));
+    expect(await readAll(output)).toEqual([new StringValue({ value: "foo" })]);
+    expect(context.responseHeader.get("Key")).toEqual("baz");
+    expect(context.responseTrailer.get("TKey")).toEqual("tbar");
+  });
+  it("should apply interceptors to bidi streaming calls", async () => {
+    const context = createHandlerContext({
+      ...handlerInit,
+      method: TestService.methods.unary,
+    });
+    context.values.set(kFoo, "bar");
+    const output = transformInvokeImplementation(
+      createMethodImplSpec(
+        TestService,
+        TestService.methods.bidiStreaming,
+        // eslint-disable-next-line @typescript-eslint/require-await
+        async function* (_, { responseHeader, responseTrailer }) {
+          responseHeader.set("Key", "bar");
+          responseTrailer.set("TKey", "tbar");
+          yield { value: "foo" };
+        },
+      ) as unknown as MethodImplSpec<Int32Value, StringValue> & {
+        kind: MethodKind.BiDiStreaming;
+      },
+      context,
+      [
+        (next) => async (req) => {
+          expect(req.stream).toEqual(true);
+          expect(req.init.method).toEqual("POST");
+          expect(req.service).toEqual(TestService);
+          expect(req.header.get("Key")).toEqual("Value");
+          expect(req.url).toEqual("https://example.com/foo");
+          expect(req.method).toEqual(TestService.methods.bidiStreaming);
+          expect(context.values.get(kFoo)).toEqual("bar");
+          req.header.set("Key", "bar");
+          const res = await next(req);
+          const responses = [];
+          for await (const next of res.message as AsyncIterable<StringValue>) {
+            responses.push(next);
+          }
+          expect(res.stream).toEqual(true);
+          expect(res.service).toEqual(TestService);
+          expect(res.header.get("Key")).toEqual("bar");
+          expect(res.method).toEqual(TestService.methods.bidiStreaming);
+          res.header.set("Key", "baz");
+          return {
+            ...res,
+            message: createAsyncIterable(responses),
+          } as typeof res;
+        },
+        (next) => async (req) => {
+          expect(req.header.get("Key")).toEqual("bar");
+          return next(req);
+        },
+      ],
+    )(createAsyncIterable([new Int32Value({ value: 1 })]));
+    expect(await readAll(output)).toEqual([new StringValue({ value: "foo" })]);
+    expect(context.responseHeader.get("Key")).toEqual("baz");
+    expect(context.responseTrailer.get("TKey")).toEqual("tbar");
+  });
+});
+
+async function readAll<T>(it: AsyncIterable<T>) {
```

Comment:
It's a tiny function, but I think it's still worth importing the existing one from "./async-iterable-helper.spec.js", it's already done at other places.